cmd/notify-mailer/main.go

```diff
@@ -231,87 +254,138 @@ func emailsForReg(id int, dbMap dbSelector) ([]string, error) {
 		return nil, err
 	}
 
-	var contactFields []string
-	var addresses []string
-	err = json.Unmarshal(contact.Contact, &contactFields)
+	var contactContents []string
+	err = json.Unmarshal(result.Contact, &contactContents)
 	if err != nil {
 		return nil, err
 	}
-	for _, entry := range contactFields {
+
+	var addresses []string
+	for _, entry := range contactContents {
 		if strings.HasPrefix(entry, "mailto:") {
 			addresses = append(addresses, strings.TrimPrefix(entry, "mailto:"))
 		}
 	}
 	return addresses, nil
 }
 
-// recipient represents one line in the input CSV, containing an account and
-// (optionally) some extra fields related to that account.
+// recipient represents one entry of recipient list file.
 type recipient struct {
-	id    int
+	id int64
+
+	// Extra is exported so the contents can be accessed to fill template
+	// variables referenced in the email body.
 	Extra map[string]string
 }
 
-// emailToRecipientMap maps from an email address to a list of recipients with
-// that email address.
-type emailToRecipientMap map[string][]recipient
+// addressesToRecipientsMap maps from an email address to a list of `recipient`s
+// with the same email address.
+type addressesToRecipientsMap map[string][]recipient
 
-// readRecipientsList reads a CSV filename and parses that file into a list of
-// recipient structs. It puts any columns after the first into a per-recipient
-// map from column name -> value.
-func readRecipientsList(filename string) ([]recipient, error) {
-	f, err := os.Open(filename)
+// parseRecipientList parses the contents of a recipient list file into a list
+// of `recipient` objects. All columns after `id` into a per-recipient map from
+// column name -> value.
+func parseRecipientList(recipientList *csv.Reader) ([]recipient, error) {
+	entries, err := recipientList.Read()
 	if err != nil {
 		return nil, err
 	}
-	reader := csv.NewReader(f)
-	record, err := reader.Read()
-	if err != nil {
-		return nil, err
-	}
-	if len(record) == 0 {
-		return nil, fmt.Errorf("no entries in CSV")
-	}
-	if record[0] != "id" {
-		return nil, fmt.Errorf("first field of CSV input must be an ID.")
+
+	if entries[0] != "id" {
+		return nil, fmt.Errorf(
+			"the first column of the header row must be \"id\", got: %q", entries[0])
 	}
-	var columnNames []string
-	for _, v := range record[1:] {
-		columnNames = append(columnNames, strings.TrimSpace(v))
+
+	// Collect column names used for the `Extra` field of the `recipient`
+	// object. All columns after `id` are considered `Extra`.
+	header := entries[1:]
+	var extraColNames []string
```

Comment:
`extraColNames` seems like a bad rename. From the perspective of this point in the program, they are just column names.

Arguably, calling the field on `recipient` "Extra" was a mistake and should be something more like "Data".